CLAUDE.md

```diff
@@ -0,0 +1,152 @@
+# Beyond All Reason - Codebase Documentation
+
+**Repository:** https://github.com/beyond-all-reason/Beyond-All-Reason
+**Game Engine:** Recoil (fork of Spring RTS Engine)
+
+---
+
+## Overview
+
+Beyond All Reason (BAR) is an open-source RTS game built on the Recoil engine.
+
+**Primary Language:** Lua
+**License:** GNU GPL v2+ (code), various Creative Commons licenses (assets - see LICENSE.md)
+
+**Components:**
+- **Game Code** (this repository) - Game logic, units, UI
+- **Lobby/Launcher** ([BYAR-Chobby](https://github.com/beyond-all-reason/BYAR-Chobby))
+- **Recoil Engine** (https://github.com/beyond-all-reason/spring)
+
+---
+
+## Architecture
+
+### Lua Execution Contexts
+
+BAR uses the Spring/Recoil engine's Lua execution model with separate contexts:

```

Comment:
This lists two major contexts, one approximately irrelevant one, one hallucinated, and skips the defs parsing context which is major.